
Calculator.java

package com.niit.junit;

public class Calculator {
public double add(double a,double b) {
	return a+b;
	
}
public double subtract(double a,double b) {
	return a-b;
	
}
public double multiply(double a, double b) {
	return a*b;
}
public double divide(double a, double b) {
	if(b==0) {
		throw new ArithmeticException();
		
	}
	return a/b;
}
}
**********************************************************************
CalculatorTest
package com.niit.junit;

import static org.junit.Assert.*;
import org.junit.After;
import org.junit.Test;
import org.junit.Before;
public class CalculatorTest {
private Calculator calc;
	@Before
	
	public void setUp() throws Exception{
		calc = new Calculator();
		
	}
	@After
	public void tearDown() throws Exception{
		calc = null;
		
	}	@Test
	public void testadd(){
		double result = calc.add(10.5, 20.32);
		assertEquals(30.82,result,0);
	}
	@Test
	public void testsubtract(){
		double result = calc.subtract(100.5,100);
		assertEquals(0.5,result,0);
	}
	@Test
	public void testmultiply(){
		double result = calc.multiply(2.5, 100);
		assertEquals(250,result,0);
	}
	@Test
	public void testdivide(){
		double result = calc.divide(100,0);
		assertEquals(10,result,0.001);
	}
	@Test(expected = ArithmeticException.class)
	public void testDivideByZero(){
		calc.divide(100.5, 0);
	}
}
